src/fileio.cpp

```diff
@@ -51,85 +50,99 @@ const char* LoadingError::what() const noexcept
- * @return Pointer to name of next entry (as a file path suitable for opening a file). Returns \c nullptr if not next entry exists.
- * @note The memory returned is owned by the #DirectoryReader object, and should not be released.
+ * Scan a given directory and return all file entries.
+ * @param Directory to scan.
  */
+std::vector<std::string> GetAllFileEntries(const std::string &path)
+{
+    std::vector<std::string> entries;
+    if(!PathIsDirectory(path)) return entries;
+
+    for(const auto &entry : std::filesystem::directory_iterator(path)) {
+        if(entry.is_regular_file()) {
+            std::string entryName = entry.path().string();
+            entries.push_back(entryName);
+        }
+    }
+    return entries;
+}
 
-/**
- * @fn void DirectoryReader::ClosePath()
- * Denote no further interest in reading the contents of the current directory.
- */
 
 /**
- * @fn const char *DirectoryReader::MakePath(const char *directory, const char *fname)
- * Construct a path from a directory and a file name.
- * @param directory Directory part of the path.
- * @param fname File name part of the path.
- * @return The combined path to the file.
+ * Delete the given file.
+ * @param path Path to delete.
  */
+void RemoveFile(const std::string &path)
+{
 
-/**
- * @fn bool DirectoryReader::EntryIsFile()
- * Test whether the last returned entry from #NextEntry is a file.
- * @return Whether the entry is a file.
- */
+	if (!PathIsFile(path)) return;
+	try {
+	    std::filesystem::remove(path);
+	} catch(std::filesystem::filesystem_error const& ex) {
+	        error("%s", ex.what());
+	}
 
-/**
- * @fn bool DirectoryReader::EntryIsDirectory()
- * Test whether the last returned entry from #NextEntry is a directory.
- * @return Whether the entry is a directory.
- */
+}
 
 /**
- * Get the next file entry.
- * Pulls entries from #NextEntry, until the end is reached or until a file is returned.
- * @return Next file entry, if available.
+ * Test whether the given path points to a normal file.
+ * @param path Path to investigate.
+ * @return Whether the given path points to a normal file.
  */
-const char *DirectoryReader::NextFile()
+bool PathIsFile(const std::string &path)
 {
-	const char *entry;
+	    std::filesystem::path fs(path);
+        try {
+	        return std::filesystem::exists(fs) && std::filesystem::is_regular_file(fs);
+	    } catch(std::filesystem::filesystem_error const& ex) {
+	        error("%s", ex.what());
+	        return false;
+	    }
 
-	do {
-		entry = this->NextEntry();
-	} while (entry != nullptr && !this->EntryIsFile());
-	return entry;
 }
 
-
 /**
- * Construct a directory reader object (specific for the operating system).
- * @return A directory reader.
- * @ingroup fileio_group
+ * Test whether the given path points to a directory.
+ * @param path Path to investigate.
+ * @return Whether the given path points to a directory.
  */
-DirectoryReader *MakeDirectoryReader()
+bool PathIsDirectory(const std::string &path)
```

Comment:
You're putting the ampersand next to the type in some places, and next to the name in other places. I'm not sure how it's done in the code at the moment, but I think it would be good to be consistent about it.

Reply:
This should be fixed now